cli/src/semdep/parse_lockfile.py

```diff
@@ -491,6 +572,55 @@ def parse_dep(s: str) -> FoundDependency:
     yield from (parse_dep(dep) for dep in deps)
 
 
+def parse_requirements(
+    lockfile_text: str, manifest_text: Optional[str]
+) -> Iterator[FoundDependency]:
+    for op in ["<", "<=", ">", ">="]:
+        if op in lockfile_text:
+            raise SemgrepError("requirements.txt contains non-pinned versions")
```

Comment:
Is it fine to raise here or will semgrep now fail on any repo with a requirements.txt file with <=

Probably worth adding a test that we dont crash with the common requirements.txt

Reply:
This will get caught by the try/except clauses in `parse_lockfile_str`. So we'll print a warning to the command line and then consider the offending `requirements.txt` to be empty, and it will produce no findings.